### PR TITLE
Update to the Features list, refresh of dependencies in package.json

### PR DESCRIPTION
**Pull request by tlberglund**
There are two things going on in this pull request. Apologies for the slight lack of discipline entailed in combining these two things in one PR:

- A comprehensive upgrade of all dependencies in `package.json` to the latest version of each as of this date. Previously, the site would not build on ARM Macs, but the up-to-date dependencies do.
- A refresh of the Features list on the home page to get better-aligned with the state of modern Pinot.

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -28,11 +28,11 @@ module.exports = {
   themeConfig: {
     metadata: [
       {name: 'description', content: 'Apache Pinot'},
-      {name: 'description', content: 'Real-time Analytics'},
-      {name: 'description', content: 'User-facing Analytics'},
+      {name: 'description', content: 'real-time Analytics'},
+      {name: 'description', content: 'user-facing Analytics'},
       {name: 'description', content: 'user facing analytics'},
```
Comment (Jackie-Jiang) [inline website/docusaurus.config.js]: Should it all be capitalized?

Comment (tlberglund) [reply website/docusaurus.config.js]: Good catch. Consistent title case is best!

**Comment by tlberglund** [comment]
Okay, I think this PR is in decent shape now. Please build the site locally and make sure you agree with the claims in the Features list before approving the PR. I've tried to be careful with it, but more 👀  are better.

**Review by xiangfu0 — DISMISSED** [review]

**Comment by xiangfu0** [comment]
Please fix the CI

**Comment by tlberglund** [comment]
> Please fix the CI

Ah yes. We'll get on that. Sorry I missed it!

**Comment by tlberglund** [comment]
@xiangfu0, do you have an email from [November, 2021 from Algolia/DocSearch](https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration) with a new appId and apiKey? The build is failing on this error.

```
➜  website git:(features-update) ✗ yarn run build
yarn run v1.22.19
$ docusaurus build
[INFO] [en] Creating an optimized production build...
[ERROR] Unable to build website for locale en.
[ERROR] ValidationError: "algolia.appId" is required. If you haven't migrated to the new DocSearch infra, please refer to the blog post for instructions: https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration
[INFO] Docusaurus version: 2.4.1
Node version: v20.5.1
error Command failed with exit code 1.```

**Review by mayankshriv — DISMISSED** [review]

**Comment by xiangfu0** [comment]
> @xiangfu0, do you have an email from [November, 2021 from Algolia/DocSearch](https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration) with a new appId and apiKey? The build is failing on this error.
> 
> ```
> ➜  website git:(features-update) ✗ yarn run build
> yarn run v1.22.19
> $ docusaurus build
> [INFO] [en] Creating an optimized production build...
> [ERROR] Unable to build website for locale en.
> [ERROR] ValidationError: "algolia.appId" is required. If you haven't migrated to the new DocSearch infra, please refer to the blog post for instructions: https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration
> [INFO] Docusaurus version: 2.4.1
> Node version: v20.5.1
> error Command failed with exit code 1.```
> ```

Fixed.
Still there are more issues related to `docusaurus build`

**Comment by tlberglund** [comment]
Refreshing the dependency list proved to be too deep a rabbit hole. We'll keep it as-is. It might be advisable to rebuild this site with a simpler stack, since it's just static content and a blog. I'll propose that in the appropriate place later on. :)

**Pull request closed without merging**